plansys2_executor/src/plansys2_executor/ExecutorClient.cpp

```diff
@@ -23,67 +23,146 @@
 namespace plansys2
 {
 
+using namespace std::chrono_literals;
+using namespace std::placeholders;
+
 using ExecutePlan = plansys2_msgs::action::ExecutePlan;
 
 ExecutorClient::ExecutorClient(rclcpp::Node::SharedPtr provided_node)
-: node_(provided_node), finished_(false)
+: node_(provided_node)
 {
-  this->execute_plan_client_ptr_ = rclcpp_action::create_client<ExecutePlan>(
-    node_->get_node_base_interface(),
-    node_->get_node_graph_interface(),
-    node_->get_node_logging_interface(),
-    node_->get_node_waitables_interface(),
-    "execute_plan");
+  createActionClient();
+}
+
+void
+ExecutorClient::createActionClient()
+{
+  action_client_ = rclcpp_action::create_client<ExecutePlan>(node_, "execute_plan");
+
+  if (!this->action_client_->wait_for_action_server(3s)) {
+    RCLCPP_ERROR(node_->get_logger(), "Action server not available after waiting");
+  }
 }
 
 bool
-ExecutorClient::executePlan()
+ExecutorClient::start_plan_execution()
 {
-  using namespace std::placeholders;
+  if (!executing_plan_) {
+    createActionClient();
+    auto success = on_new_goal_received();
+
+    if (success) {
+      executing_plan_ = true;
+      return true;
+    }
+  } else {
+    RCLCPP_INFO(node_->get_logger(), "Already executing a plan");
+  }
 
-  finished_ = false;
+  return false;
+}
+
+bool
+ExecutorClient::execute_and_check_plan()
+{
+  if (rclcpp::ok() && !goal_result_available_) {
+    rclcpp::spin_some(node_);
 
-  if (!this->execute_plan_client_ptr_) {
-    RCLCPP_ERROR(node_->get_logger(), "Action client not initialized");
+    if (!goal_result_available_) {
+      return true;  // Plan not finished
```

Comment:
This seems to contradict your description in the PR discussion. ...

"This method must be called iteratively to refresh the result and feedback. It returns true if the plan execution finishes."

Should this return be false and the other true?

Reply:
Yes, you are right. I noticed this yesterday when preparing the experiment. It returns true while the plan is still executing.

I will edit the PR description unless you think it's more appropriate to return true when finishes.

Reply:
No. I think it's fine the way you have it. Returning true to indicate that it's still executing makes sense to me. Just double checking that this was the intended behavior.